### PR TITLE
Split each OCW run into its own course

### LINKED ISSUE #3243: Each OCW run should be a distinct course
Currently, all OCW `LearningResourceRuns` with the same course number are grouped under the same `Course`.  Instead, each run should be treated as a separate course.  

- Write a function to split existing runs into separate courses and update the index.
- Modify the OCW import function to create 1 course per run.
- Modify the `course_id` of OCW courses to be a combination of run id and course number.

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #3243 

#### What's this PR do?
Adds a management command to split each OCW run into its own course.  Modifies OCW import functions to do the same.

#### How should this be manually tested?
- On the master branch, import the following if you don't already have it:
```
from course_catalog.api import *
sync_ocw_courses(course_prefixes=[
"PROD/18/18.01/Fall_2003/18-01-single-variable-calculus-fall-2003/",
"PROD/18/18.01/Fall_2005/18-01-single-variable-calculus-fall-2005/",
"PROD/18/18.01/Fall_2006/18-01-single-variable-calculus-fall-2006/",
], blocklist=[], force_overwrite=True, upload_to_s3=True)
```
- Search for "18.01" in open discussions, there should be 1 result for "Single Variable Calculus" and it should have 2 published runs.
- Run `python manage.py split_ocw_courses`
- Run `python manage.py recreate_index`
- Search for "18.01" in open discussions, there should be 2 results for "Single Variable Calculus" and they should each have 1 published run.  Thumbnails should be different.

- Run `backpopulate_ocw_courses` for a little while, or run the `sync_ocw_courses` function above again on a subset of prefixes, and make sure it works correctly

### CODE REVIEW

File: course_catalog/management/commands/split_ocw_courses.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+"""Management command for splitting OCW courses by run"""
+
+from django.core.management import BaseCommand
+from djoser.conf import settings
```
Comment (abeglova) [inline course_catalog/management/commands/split_ocw_courses.py:4]: Should this be `from django.conf import settings`?

Comment (mbertrand) [reply course_catalog/management/commands/split_ocw_courses.py:4]: Yes, and I remember changing it yesterday afternoon, must have reverted again somehow, oops.  I'm going to blame PyCharm 😉

File: course_catalog/api.py [hunk 2]
```diff
@@ -63,6 +64,50 @@ def safe_load_json(json_string, json_file_key):
         return {}
 
 
+def split_ocw_courses_by_run(chunk_size=settings.OCW_ITERATOR_CHUNK_SIZE):
+    """
+    Create a separate course for each OCW run.
+    A reindex will be necessary afterward.
+    """
+    ocw = PlatformType.ocw.value
+    for course in Course.objects.filter(platform="ocw").iterator(chunk_size):
```
Comment (abeglova) [inline course_catalog/api.py:73]: Filter for published here too?

File: scripts/test/js_test.sh [hunk 3]
```diff
@@ -51,6 +51,7 @@ if [[ $(
     grep -v "Accessing PropTypes" |
     grep -v "Accessing createClass" |
     grep -v "Browserslist: caniuse-lite is outdated" |
+    grep -v "npx browserslist@latest" |
```
Comment (abeglova) [inline scripts/test/js_test.sh:54]: Why was this change necessary?

File: pylintrc [hunk 4]
```diff
@@ -19,4 +19,4 @@ ignored-modules=
 	six.moves,
 
 [MESSAGES CONTROL]
-disable = no-member, old-style-class, no-init, too-few-public-methods, abstract-method, invalid-name, too-many-ancestors, line-too-long, no-self-use, len-as-condition, no-else-return, cyclic-import, duplicate-code, inconsistent-return-statements, bad-continuation,
+disable = no-member, old-style-class, no-init, too-few-public-methods, abstract-method, invalid-name, too-many-ancestors, line-too-long, no-self-use, len-as-condition, no-else-return, cyclic-import, duplicate-code, inconsistent-return-statements, bad-continuation, c-extension-no-member
```
Comment (abeglova) [inline pylintrc:22]: Why was this change necessary?

File: course_catalog/api.py (hunk 2 above)
Comment (mbertrand) [reply course_catalog/api.py:73]: I think the unpublished courses should be processed for consistency (new course_id, 1 run each).

**Review by abeglova — CHANGES_REQUESTED** [review]
Looks good! Back to you with minor comments

File: scripts/test/js_test.sh (hunk 3 above)
Comment (mbertrand) [reply scripts/test/js_test.sh:54]: Fixes test errors on github (https://github.com/mitodl/open-discussions/pull/3247)

**Review by abeglova — APPROVED** [review]
👍
